ui/appui-react/src/appui-react/frontstage/FrontstageDef.tsx

```diff
@@ -113,7 +115,9 @@ export class FrontstageDef {
     return this._floatingContentControls;
   }
 
-  public get toolSettings(): WidgetDef | undefined {
+  public get toolSettings():
```

Comment:
This changes the type of a public API, just add separate fields and internal properties if needed. Alternatively, just add a getter for `_initialConfig` (just make sure to tag it as `@internal`).

Reply:
Split the method into toolSettings method and activeToolbarEmptyNode methods.